### PR TITLE
1.0.2/agenda item create task offline

### PR DESCRIPTION
**Pull request by juandgaines**
- update agenda item reminder/task  and scheduled if fails
- delete agenda item reminder/task  and scheduled if fails
- create agenda item reminder/task  and scheduled if fails

### CODE REVIEW

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/AgendaPendingSyncScheduler.kt [hunk 1]
```diff
@@ -121,6 +135,60 @@ class AgendaPendingSyncScheduler (
...
+            .build()
+        applicationScope.launch {
+            workManager.enqueue(workRequest).await()
+        }.join()
+    }
+
+    private suspend fun scheduleCreateAgendaItem(agendaItem: AgendaItems) {
+        val userId = sessionStorage.get()?.userId ?: return
+
+        val type = when (agendaItem) {
```
Comment (philipplackner) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/AgendaPendingSyncScheduler.kt:150]: Could it make sense to introduce an enum for the type, so it's exclusive?

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/workers/UpdateAgendaItemWorker.kt [hunk 2]
```diff
@@ -16,33 +19,51 @@ class UpdateAgendaItemWorker @AssistedInject constructor(
...
+                    )
+
+                    return if (response is com.juandgaines.core.domain.util.Result.Error) {
+                        response.error.toWorkerResult()
+                    } else {
+                        agendaSyncDao.deleteUpdateReminderSync(updatedPendingAgendaItem.reminderId)
+                        Result.success()
+                    }
+                }?:Result.failure()
             }
```
Comment (philipplackner) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/workers/UpdateAgendaItemWorker.kt:60]: Something to account for for all update operations is the edge case when the user is offline, creates an agenda item (not synced yet), then updates it when it's not synced. You could run into a race condition where once the user gets back online, the update worker triggers before the create worker leading to inconsistent data.

To fix that, you'll need to check if there is already a pending create data for that agenda item and if so do that before the update.

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt [hunk 3]
```diff
@@ -283,15 +320,18 @@ fun AgendaItemScreen(
...
-                Text(
-                    text = stringResource(id = R.string.delete_item, agendaItemName).uppercase(),
-                    style = MaterialTheme.typography.titleSmall,
-                    color = TaskyGray,
-                    textAlign = TextAlign.Center,
-                    modifier = Modifier
-                        .padding(16.dp)
-                        .fillMaxWidth()
-                )
+                if(state.isNew.not()){
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt:323]: Do you find this more readable than simply `!state.isNew`?

**Review by philipplackner — APPROVED** [review]

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/workers/UpdateAgendaItemWorker.kt (hunk 2 above)
Comment (juandgaines) [reply agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/workers/UpdateAgendaItemWorker.kt:60]: awesome, i covered it thanks for the feedback

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt (hunk 3 above)
Comment (juandgaines) [reply agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt:323]: yes you are right

**Pull request merged**